Github Action to tweet when labeled with tag "issue/tweet"

gopivaibhav opened the pull request:
Signed-off-by: Gopi Vaibhav <gopivaibhav0@gmail.com>

**Notes for Reviewers**

We have to add these keys in the secrets in order to make this Github Action work:
- TWITTER_CONSUMER_API_KEY
- TWITTER_CONSUMER_API_SECRET
- TWITTER_ACCESS_TOKEN
- TWITTER_ACCESS_TOKEN_SECRET 

Video Proof of working after adding these secrets -
[Screen Recording by adding different tags to test](https://user-images.githubusercontent.com/78922463/209333896-76b25d3f-48d7-4f7f-bc8e-65a333ddf665.webm)


This PR fixes #6612 


**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.
 

<!--
Thank you for contributing to Meshery! 

Contributing Conventions:

1. Include descriptive PR titles with [<component-name>] prepended.
2. Build and test your changes before submitting a PR. 
3. Sign your commits

By following the community's contribution conventions upfront, the review process will 
be accelerated and your PR merged more quickly.
-->

gopivaibhav commented:
Haii @leecalcote I have made the PR (including the demo video of sending a tweet from my account) with the required changes as discussed in the yesterday's meet but I am not sure of what I have to add in the tweet. So, I have added a demo message as of now and will change it according to the desired change.

gopivaibhav commented:
@leecalcote I have made the commit which restricts the action to run only when members of meshery adds the `issue/tweet` label.
Video demo of the sample tweet with issue title and URL-
[Screencast from 12-25-2022 12:22:33 AM.webm](https://user-images.githubusercontent.com/78922463/209448236-ad93d5e9-2b9b-4ae2-97b3-ba051920863c.webm)

ayushthe1 reviewed (commented):
hey in the issue it's mentioned that the workflow should run when label is applied either to issue or a pull request. So i think you should also add the support for pull request.

leecalcote commented on .github/workflows/label-tweet.yml:
```diff
@@ -0,0 +1,31 @@
+    if: ${{ github.event.label.name == 'issue/tweet' }}
+    runs-on: ubuntu-latest
+    steps:
+        - name: Check if organization member
```
@gopivaibhav more than this, will you try using this - action https://github.com/tspascoal/get-user-teams-membership - to assess whether the user is a member of @meshery/maintainers?

gopivaibhav replied:
Oh, okay. I will change it. Also, What am I supposed to tweet?
Is the content which I have written now looking good? Or do I have to change it to something else?

leecalcote replied:
I missed your message here. Let's try this:

```
Help wanted: "<issue title>" - <issue link> #opensource
```

gopivaibhav replied:
Added the required tweet body for the tweet

leecalcote commented:
Some good stuff in here, @gopivaibhav

Chadha93 commented:
@gopivaibhav  Let's discuss the update of this PR on the Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

gopivaibhav commented:
> @gopivaibhav  Let's discuss the update of this PR on the Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

Sorry @Chadha93 I have been writing my exams for 5 days. I will be back again from 9th of January and will add this to the agenda of Meshery dev call then.

gopivaibhav commented:
Hi @leecalcote @ayushthe1 I was busy with my exams. So, couldn't work more but I have updated the code to work for both issues and PRs. I also restricted only maintainers to tweet when labeled with `issue/tweet`. Let me know If I have to change anything.


Note - 
- We have to add a secret with the name PAT, which has the value of the GitHub token of this org with the scope- `read:org`
- I have added "maintainers" as the team name in lowercase but if the team name is "Maintainers", my code will result in a false for checking whether the actor is a maintainer or not. So, let me know the proper spelling of the team name with the correct case as I can't view the maintainers' team.

leecalcote commented:
New org-wide secrets defined for consumer-key and consumer-secret.

leecalcote commented:
> Hi @leecalcote @ayushthe1 I was busy with my exams. So, couldn't work more but I have updated the code to work for both issues and PRs. I also restricted only maintainers to tweet when labeled with `issue/tweet`. Let me know If I have to change anything.
> 
> Note -
> 
> * We have to add a secret with the name PAT, which has the value of the GitHub token of this org with the scope- `read:org`
> * I have added "maintainers" as the team name in lowercase but if the team name is "Maintainers", my code will result in a false for checking whether the actor is a maintainer or not. So, let me know the proper spelling of the team name with the correct case as I can't view the maintainers' team.

Great notes, @gopivaibhav. Thank you. I've shifted us from `PAT` to `GLOBAL_TOKEN` and have two of the four keys loaded thus far.

Nikhil-Ladha commented on .github/workflows/label-tweet.yml:
```diff
@@ -0,0 +1,43 @@
+        if: ${{ github.event.pull_request && steps.checkUserMember.outputs.isTeamMember == 'true' }}
+        uses: ethomson/send-tweet-action@v1
+        with:
+          status: 'Help wanted on thi PR: "${{ github.event.pull_request.title }}" - ${{ github.event.pull_request.html_url }} \#opensource'
```
nit: this

ayushthe1 commented:
hey @gopivaibhav ,now that your pr is almost complete ,could you also review mine at [here](https://github.com/layer5io/layer5/pull/3556) .

Chadha93 commented:
@gopivaibhav Let's discuss the update of this PR on the Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

Nikhil-Ladha reviewed (dismissed):
lgtm

Nikhil-Ladha commented:
Leaving the final call on @leecalcote to confirm the verbiage.

gopivaibhav commented:
> hey @gopivaibhav ,now that your pr is almost complete ,could you also review mine at [here](https://github.com/layer5io/layer5/pull/3556) .

I have gone through your PR @ayushthe1 And it's more or less the same as the code I wrote. So, I guess there's nothing to review

leecalcote commented on .github/workflows/label-tweet.yml:
```diff
@@ -0,0 +1,43 @@
+        if: ${{ github.event.issue && steps.checkUserMember.outputs.isTeamMember == 'true' }}
+        uses: ethomson/send-tweet-action@v1
+        with:
+          status: 'Help wanted on this issue: "${{ github.event.issue.title }}" - ${{ github.event.issue.html_url }} \#opensource'
```
```suggestion
          status: 'Help Wanted: "${{ github.event.issue.title }}" - ${{ github.event.issue.html_url }} #opensource'
```

leecalcote commented on .github/workflows/label-tweet.yml:
```diff
@@ -0,0 +1,43 @@
+        if: ${{ github.event.pull_request && steps.checkUserMember.outputs.isTeamMember == 'true' }}
+        uses: ethomson/send-tweet-action@v1
+        with:
+          status: 'Help wanted on this PR: "${{ github.event.pull_request.title }}" - ${{ github.event.pull_request.html_url }} \#opensource'
```
```suggestion
          status: 'Help Wanted: "${{ github.event.pull_request.title }}" - ${{ github.event.pull_request.html_url }} #opensource'
```

leecalcote commented:
> Leaving the final call on @leecalcote to confirm the verbiage.

Thank you, @Nikhil-Ladha. I've tweaked the verbiage on each tweet. ✅ 

We're without two of the necessary secrets just yet: API Token and Token Secret. Work to be done here.

gopivaibhav commented:
Hi @leecalcote I am back to being active as I don't have any exams now. So, can you please let me know the status of this PR or if You are expecting anything else from me?

Chadha93 commented:
@gopivaibhav Let's discuss the update of this PR on the Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

gopivaibhav commented:
Hi @leecalcote Any update on this PR?
